.github/actions/detect-workflow/action.yml

```diff
@@ -0,0 +1,31 @@
+name: 'Detect Reusable Workflow Repository and Ref'
+description: 'Get the current reusable workflow repository name and ref. Requires "id-token: write" permission'
+outputs:
+  repository:
+    description: The current workflow repository, format org/repository
+    value: ${{ steps.detect.outputs.repository }}
+  ref:
+    description: The current workflow reference
+    value: ${{ steps.detect.outputs.ref }}
+runs:
+  using: 'composite'
+  steps:
+    - run: | 
+        set -euo pipefail
```

Comment:
I thought that maybe we could use the Go code to do this since we have retrieval, parsing, and verification in the [github](https://github.com/slsa-framework/slsa-github-generator/tree/main/github) package. Though I realize it would be a bit more involved to compile and run it and shell is convenient.

Reply:
updated to use this! was extremely easy PTAL

Reply:
Good to hear! Kudos for doing this.